### PR TITLE
fix: use proper ALPN protocols if HTTP client is HTTP/1.1 only

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/16923

### CODE REVIEW

**Review by levex — APPROVED** [review]
Makes sense

File: ext/node/polyfills/http.ts [hunk 1]
```diff
@@ -641,6 +641,7 @@ class ClientRequest extends OutgoingMessage {
         }
         this._client.close();
         const incoming = new IncomingMessageForClient(this.socket);
+        incoming.req = this;
```
Comment (bartlomieju) [inline ext/node/polyfills/http.ts:644]: We should do `req.res = incoming;` too

**Comment by mmastrac** [comment]
Looks like the Windows test is failing because Windows surfaces a connection reset early before allowing you to read the remainder of the buffer. The test server would need to gracefully close the TLS connection (ideally sending the CloseNotify) on failure to avoid this.

File: cli/tests/unit_node/http_test.ts [hunk 2]
```diff
@@ -509,3 +510,24 @@ Deno.test("[node/http] ClientRequest handle non-string headers", async () => {
   await def;
   assertEquals(headers!["1"], "2");
 });
+
+Deno.test("[node/http] ClientRequest use HTTP/1.1", async () => {
+  let body = "";
```
Comment (mmastrac) [inline cli/tests/unit_node/http_test.ts:515]:
```suggestion
Deno.test("[node/http] ClientRequest uses HTTP/1.1", async () => {
  let body = "";
```

**Review by mmastrac — APPROVED** [review]
LGTM, one typo spotted

**Pull request merged**